viper.go

```diff
@@ -1162,34 +1168,38 @@ func (v *Viper) InConfig(key string) bool {
 }
 
 // SetDefault sets the default value for this key.
-// SetDefault is case-insensitive for a key.
+// SetDefault's case-sensitivity for key is determined by viper.keyCaseSensitivity.
 // Default only used when no value is provided by the user via flag, config or ENV.
 func SetDefault(key string, value interface{}) { v.SetDefault(key, value) }
 func (v *Viper) SetDefault(key string, value interface{}) {
 	// If alias passed in, then set the proper default
-	key = v.realKey(strings.ToLower(key))
-	value = toCaseInsensitiveValue(value)
+	key = v.realKey(v.caseKey(key))
+	if !v.keyCaseSensitivity {
+		value = toCaseInsensitiveValue(value)
+	}
 
 	path := strings.Split(key, v.keyDelim)
-	lastKey := strings.ToLower(path[len(path)-1])
+	lastKey := v.caseKey(path[len(path)-1])
 	deepestMap := deepSearch(v.defaults, path[0:len(path)-1])
 
 	// set innermost value
 	deepestMap[lastKey] = value
 }
 
 // Set sets the value for the key in the override register.
-// Set is case-insensitive for a key.
+// Set's case-sensitivity for a key is determined by viper.keyCaseSensitivity.
```

Comment:
I would retain the original comment and add something like this:
`This behavior can be modified by viper.SetKeyCaseSensitivity`